src/components/dar_drafts_table/DarDraftTable.js

```diff
@@ -0,0 +1,254 @@
+          justifyContent: 'left',
+        },
+        key: `draft-buttons-cell-${id}`,
+      },
+      [resumeButtonTemplate, deleteButtonTemplate]
+    ),
+  };
+};
+
+//helper function that foramts table metadata for cells row by row, returns an array of arrays
+const processDraftsRowData = ({visibleDrafts, showConfirmationModal, history}) => {
+  if(!isEmpty(visibleDrafts)) {
+    return visibleDrafts.map((draft) => {
+      const { id, data, createDate, updateDate } = draft;
+      const { projectTitle, partialDarCode } = !isEmpty(data) ? data : {};
+      return [
+        partialDarCodeCell({partialDarCode, id}),
+        projectTitleCell({projectTitle, id}),
+        updateDateCell({updateDate: updateDate || createDate, id}),
+        actionCellData({targetDraft: draft, showConfirmationModal, history})
+      ];
+    });
+  }
+};
+
+export default function DarDraftTable(props) {
+  const [visibleDrafts, setVisibleDrafts] = useState([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [pageCount, setPageCount] = useState(1);
+  const [tableSize, setTableSize] = useState(10);
+  const [showConfirmation, setShowConfirmation] = useState(false);
+  const [selectedDraft, setSelectedDraft] = useState({});
+
+  const { history, deleteDraft, isLoading, drafts } = props;
+
+  //sequence of events that fires when table either changes its visible window (data updates, page change, table size change, etc)
+  useEffect(() => {
+    recalculateVisibleTable({
+      tableSize,
+      pageCount,
+      filteredList: drafts,
+      currentPage,
+      setPageCount,
+      setCurrentPage,
+      setVisibleList: setVisibleDrafts
+    });
+  }, [tableSize, pageCount, currentPage, drafts]);
+
+  //callback function to change tableSize, passed to pagination bar
+  const changeTableSize = useCallback((value) => {
+    if (value > 0 && !isNaN(parseInt(value))) {
+      setTableSize(value);
+    }
+  }, []);
+
+  //function that fires on delete button press, makes confirmation modal visible and selectes target draft
+  const showConfirmationModal = (draft) => {
+    setSelectedDraft(draft);
+    setShowConfirmation(true);
+  };
+
+  //function that fires when user updates current page (buttons, manual input), passed to pagination bar
+  const goToPage = useCallback(
+    (value) => {
+      updatePage(value, pageCount, setCurrentPage);
+    },
+    [pageCount]
+  );
+
+  //helper function that formats string for modal header
+  const getModalHeader = () => {
+    if(!isEmpty(selectedDraft)) {
+      const { data, id } = selectedDraft;
+      return getIdentifier({id, data});
+    }
+  };
+
+  //delete function that fires on modal confirmation, removes draft from listing via prop delete function
+  const deleteOnClick = async() => {
+    const {id, data, referenceId} = selectedDraft;
+    const identifier = getIdentifier({id: id || referenceId, data});
+    await deleteDraft({ referenceId, identifier, });
+    setShowConfirmation(false);
+  };
```

Comment:
I don't think this isn't refreshing the underlying list of draft dars ... after delete, the deleted draft remains visible onscreen.

Reply:
Yeah, that's a bug, should be fixed with the latest commit
![Kapture 2021-11-09 at 07 40 35](https://user-images.githubusercontent.com/6664778/140926834-1f35310c-1ee2-48ca-a642-dcede5815678.gif)